src/types/to_sql.rs

```diff
@@ -87,37 +87,27 @@ pub trait ToSql {
     fn to_sql(&self) -> Result<ToSqlOutput<'_>>;
 }
 
-impl ToSql for Box<dyn ToSql> {
-    fn to_sql(&self) -> Result<ToSqlOutput<'_>> {
-        let derefed: &dyn ToSql = &**self;
-        derefed.to_sql()
-    }
-}
-
 impl<T: ToSql + Clone> ToSql for Cow<'_, T> {
     fn to_sql(&self) -> Result<ToSqlOutput<'_>> {
         self.as_ref().to_sql()
     }
 }
 
-impl<T: ToSql> ToSql for Box<T> {
+impl<T: ToSql + ?Sized> ToSql for Box<T> {
     fn to_sql(&self) -> Result<ToSqlOutput<'_>> {
-        let derefed: &dyn ToSql = &**self;
-        derefed.to_sql()
+        self.as_ref().to_sql()
     }
 }
 
 impl<T: ToSql> ToSql for std::rc::Rc<T> {
```

Comment:
Guessing we need this for Arc<T> and Rc<T> too?

Reply:
What do you mean?
`Arc` and `Rc` impls already use
```rust
self.as_ref().to_sql()
```

Reply:
I mean they should be `impl<T: ToSql + ?Sized>`

Reply:
You marked this as resolved but it's not resolved...